Classes/BAPromise.h

```diff
@@ -8,14 +8,20 @@
 
 #import <Foundation/Foundation.h>
 
+
 /* block definitions */
 typedef void (^BAPromiseOnFulfillBlock)(id _Nullable value);
 typedef void (^BAPromiseOnRejectedBlock)(NSError * _Nonnull error);
 typedef id _Nullable (^BAPromiseThenRejectedBlock)(NSError * _Nonnull error);
 typedef dispatch_block_t BAPromiseFinallyBlock;
 
+/* block runner base for promise / canceltoken */
+@interface BARunBlock : NSObject
```

Comment:
@ralph-e-boy In my view, better use more common name, we would want to add another functionality to it

Reply:
whats a good name?